mnefun/_mnefun.py

```diff
@@ -829,29 +873,125 @@ def run_sss_positions(fname_in, fname_out, host='kasga', opts='', port=22):
             fnames_in.append(next_name)
         else:
             break
-    port = str(int(port))
-    t0 = time()
-    remote_ins = ['~/' + op.basename(fname) for fname in fnames_in]
-    remote_out = '~/temp_%s_raw_quat.fif' % t0
-    remote_hp = '~/temp_%s_hp.txt' % t0
+
     print('  Copying file to %s' % host)
-    cmd = ['scp', '-P' + port] + fnames_in + [host + ':~/']
+    cmd = ['scp', '-P' + str(port)] + fnames_in + [host + ':~/']
     run_subprocess(cmd, stdout=None, stderr=None)
+    t0 = time()
+    remote_ins = ['~/' + op.basename(f) for f in fnames_in]
+    fnames_out = [op.basename(r)[:-4] + '.tmp' for r in remote_ins]
+    for fi, file_out in enumerate(fnames_out):
+        remote_out = '~/temp_%s_raw_quat.fif' % t0
+        remote_hp = '~/temp_%s_hp.txt' % t0
+
+        print('  Running maxfilter as %s' % host)
+        cmd = ['ssh', '-p', str(port), host,
+               '/neuro/bin/util/maxfilter -f ' + remote_ins[fi] + ' -o ' + remote_out +
+               ' -headpos -format short -hp ' + remote_hp + ' ' + opts]
+        run_subprocess(cmd)
 
-    print('  Running maxfilter on %s' % host)
-    cmd = ['ssh', '-p', port, host,
-           'maxfilter -f ' + remote_ins[0] + ' -o ' + remote_out +
-           ' -headpos -format short -hp ' + remote_hp + ' ' + opts]
-    run_subprocess(cmd)
-
-    print('  Copying result to %s' % fname_out)
-    cmd = ['scp', '-P' + port, host + ':' + remote_hp, fname_out]
-    run_subprocess(cmd)
+        print('  Copying result to %s' % file_out)
+        cmd = ['scp', '-P' + str(port), host + ':' + remote_hp, op.join(pout, file_out)]
+        run_subprocess(cmd)
 
-    print('  Cleaning up %s' % host)
-    cmd = ['ssh', '-p', port, host, 'rm -f %s %s %s'
+        print('  Cleaning up %s' % host)
+        cmd = ['ssh', '-p', str(port), host, 'rm -f %s %s %s'
            % (' '.join(remote_ins), remote_hp, remote_out)]
-    run_subprocess(cmd)
+        run_subprocess(cmd)
+
+    # concatenate hp pos file for split raw files if any
+    pos_data = []
+    for f in fnames_out:
+        with open(op.join(pout, f)) as infile:
+            content = infile.readlines()
+        header = content[0]
+        qdata = content[1:]
+        pos_data.append(qdata)
+        os.remove(op.join(pout, f))
+    pos_data.insert(0, header)
+
+    with open(fname_out, 'w') as fo:
+        fo.writelines(''.join(str(j) for j in i) for i in pos_data)
```

Comment:
Actually instead of messing with all this writing, you should be able to use `mne.chpi.read_head_quats` to get arrays, concatenate them along axis 0 with `np.concatenate(..., axis=0)` and then write the result with `mne.chpi.write_head_quats`